### PR TITLE
Adding thrown exceptions to JavaType.Method 

### PR DESCRIPTION
**Pull request by tkvangorder**
This PR adds exceptions to rewrite's Method types and normalizes the build methods in JavaType (which is a breaking API change)

### CODE REVIEW

File: rewrite-java-11/src/main/java/org/openrewrite/java/Java11ParserVisitor.java [hunk 1]
```diff
@@ -1537,26 +1536,38 @@ private Space statementDelim(@Nullable Tree t) {
...
             }
 
             List<String> paramNames = new ArrayList<>();
-            for (Symbol.VarSymbol p : genericSymbol.params()) {
+            for (Symbol.VarSymbol p : methodSymbol.params()) {
                 String s = p.name.toString();
                 paramNames.add(s);
             }
 
+            List<JavaType.FullyQualified> exceptionTypes = new ArrayList<>();
```
Comment (tkvangorder) [inline rewrite-java-11/src/main/java/org/openrewrite/java/Java11ParserVisitor.java:1551]: Curious if these should be JavaType.FullyQualified or JavaType.Class?

Comment (tkvangorder) [reply rewrite-java-11/src/main/java/org/openrewrite/java/Java11ParserVisitor.java:1551]: Also wasn't sure what the behavior should be when the exception class cannot be resolved. Right now, it just skips any exceptions that could not be resolved.

File: rewrite-java/src/main/java/org/openrewrite/java/tree/JavaType.java [hunk 2]
```diff
@@ -156,8 +155,8 @@ public String toString() {
 
         private final String flyweightId;
 
-        private Class(String fullyQualifiedName,
-                      int flagsBitMap,
+        private Class(int flagsBitMap,
```
Comment (tkvangorder) [inline rewrite-java/src/main/java/org/openrewrite/java/tree/JavaType.java:158]: Reordered the arguments in buildClass, buildMethod and the Variable constructors such that the flags are always the first argument.

File: rewrite-java-11/src/main/java/org/openrewrite/java/Java11ParserVisitor.java (hunk 1 above)
Comment (sambsnyd) [reply rewrite-java-11/src/main/java/org/openrewrite/java/Java11ParserVisitor.java:1551]: Re: FullyQualified vs Class - I'm not completely sure but I suspect `Class` is fine.

RE: unresolvable classes. If you know their fully qualified name I might vote to construct a synthetic type for them. If that isn't feasible then skipping is probably the thing to do

File: rewrite-java/src/main/java/org/openrewrite/java/tree/JavaType.java [hunk 3]
```diff
@@ -305,6 +304,23 @@ public static Class build(String fullyQualifiedName,
...
+        @Nullable public static Class findClass(String fullyQualifiedName) {
+            Set<Class> classes;
+           synchronized (flyweights) {
+               classes = flyweights.get(fullyQualifiedName);
+           }
+           if (classes != null) {
+               return classes.stream().findFirst().orElse(null);
+           }
+           return null;
+        }
```
Comment (sambsnyd) [inline rewrite-java/src/main/java/org/openrewrite/java/tree/JavaType.java:323]: I don't see this used anywhere. What's the use case for this?

**Review by sambsnyd — APPROVED** [review]

File: rewrite-java/src/main/java/org/openrewrite/java/tree/JavaType.java (hunk 3 above)
Comment (tkvangorder) [reply rewrite-java/src/main/java/org/openrewrite/java/tree/JavaType.java:323]: I pulled these changes out of my "build types on the fly" branch, I do use this method in that branch, I can just remove it for now.

File: rewrite-java-11/src/main/java/org/openrewrite/java/Java11ParserVisitor.java (hunk 1 above)
Comment (tkvangorder) [reply rewrite-java-11/src/main/java/org/openrewrite/java/Java11ParserVisitor.java]: Thanks for the feedback. I changed Java11ParserVisitor to create JavaType.Class for exceptions. Non-resolved instances will then have a synthetic created via JavaType.Class.build()

File: rewrite-java/src/main/java/org/openrewrite/java/tree/JavaType.java (hunk 3 above)
Comment (tkvangorder) [reply rewrite-java/src/main/java/org/openrewrite/java/tree/JavaType.java]: I have removed this method.

**Pull request merged**